_vector-search/tutorials/ai-search-flows/building-flows.md

```diff
@@ -0,0 +1,629 @@
+
+### ML resources
+
+Create and deploy an [Amazon Titan Text Embeddings model on Amazon Bedrock](https://github.com/opensearch-project/dashboards-flow-framework/blob/main/documentation/models.md#amazon-bedrock-titan-text-embedding).
+
+### Index
+
+Ensure the index settings include `index.knn: true` and your index contains a `knn_vector` field specified in the mappings as follows:
+
+```json
+{
+    "settings": {
+        "index": {
+            "knn": true
+        }
+    },
+    "mappings": {
+        "properties": {
+            "<embedding_field_name>": {
+                "type": "knn_vector",
+                "dimension": "<embedding_size>"
+            }
+        }
+    }
+}
+```
+{% include copy.html %}
+
+### Ingest pipeline
+
+Configure a single ML inference processor. Map your input text to the `inputText` model input field. Optionally,, map the output `embedding` to a new document field.
+
+### Search pipeline
+
+Configure a single ML inference search request processor. Map the query field containing the input text to the `inputText` model input field. Optionally, map the output `embedding` to a new field. Override the query to include a `knn` query, for example:
+
+```json
+{
+    "_source": {
+        "excludes": [
+            "<embedding_field>"
+        ]
+    },
+    "query": {
+        "knn": {
+            "<embedding_field>": {
+                "vector": ${embedding},
+                "k": 10
+            }
+        }
+    }
+}
+```
+{% include copy.html %}
+
+---
+
+## Hybrid search 
+
+Hybrid search combines keyword and vector search. This example demonstrates configuring hybrid search.
+
+### ML resources
+
+Create and deploy an [Amazon Titan Text Embeddings model on Amazon Bedrock](https://github.com/opensearch-project/dashboards-flow-framework/blob/main/documentation/models.md#amazon-bedrock-titan-text-embedding).
+
+### Index
+
+Ensure the index settings include `index.knn: true` and your index contains a `knn_vector` field specified in the mappings as follows:
+
+```json
+{
+    "settings": {
+        "index": {
+            "knn": true
+        }
+    },
+    "mappings": {
+        "properties": {
+            "<embedding_field_name>": {
+                "type": "knn_vector",
+                "dimension": "<embedding_size>"
+            }
+        }
+    }
+}
+```
+{% include copy.html %}
+
+### Ingest pipeline
+
+Configure a single ML inference processor. Map your input text to the `inputText` model input field. Optionally, map the output `embedding` to a new document field.
+
+### Search pipeline
+
+Configure an ML inference search request processor and a normalization processor.
+
+**For the ML inference processor**, map the query field containing the input text to the `inputText` model input field. Optionally, map the output `embedding` to a new field. Override the query to contain a `hybrid` query. Ensure to specify the `embedding_field`, `text_field`, and `text_field_input`:
```

Comment:
```suggestion
**For the ML inference processor**, map the query field containing the input text to the `inputText` model input field. Optionally, map the output `embedding` to a new field. Override the query so that it contains a `hybrid` query. Make sure to specify the `embedding_field`, `text_field`, and `text_field_input`:
```